### PR TITLE
Update 0005-linked-list.md

### PR DESCRIPTION
**Pull request by ReinierMaas**
Fixing typos.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: doc/src/challenges/0005-linked-list.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 ## Goal
 
-Verify the memory safety of [`alloc::collections::linked_list` functions](https://github.com/rust-lang/rust/blob/c290e9de32e8ba6a673ef125fde40eadd395d170/library/alloc/src/collections/linked_list.rs) that iterate the its internal inductive-defined data type.
+Verify the memory safety of [`alloc::collections::linked_list` functions](https://github.com/rust-lang/rust/blob/c290e9de32e8ba6a673ef125fde40eadd395d170/library/alloc/src/collections/linked_list.rs) that iterate its internal inductive-defined data type.
```
Comment (tautschnig) [inline doc/src/challenges/0005-linked-list.md:14]:
```suggestion
Verify the memory safety of [`alloc::collections::linked_list` functions](https://github.com/rust-lang/rust/blob/c290e9de32e8ba6a673ef125fde40eadd395d170/library/alloc/src/collections/linked_list.rs) that iterate over its internal inductive-defined data type.
```

**Review by tautschnig — APPROVED** [review]
Thank you for catching this, but I think we still need a further fix here as indicated below.

File: doc/src/challenges/0005-linked-list.md (hunk 1 above)
Comment (ReinierMaas) [reply doc/src/challenges/0005-linked-list.md]: Thanks for thinking along!

**Comment by havelund** [comment]
Just a typo which has been corrected. Ok.

**Review by feliperodri — APPROVED** [review]
LGTM.

**Pull request merged**